internal/driver/glfw/menu_darwin.go

```diff
@@ -139,11 +139,35 @@ func exceptionCallback(e *C.char) {
 
 func handleSpecialItems(w *window, menu *fyne.Menu, nextItemID int, addSeparator bool) (*fyne.Menu, int) {
 	for i, item := range menu.Items {
-		if item.Label == "Settings" || item.Label == "Settings…" || item.Label == "Preferences" || item.Label == "Preferences…" {
+		switch item.Label {
+		case "Settings", "Settings…", "Preferences", "Preferences…":
 			items := make([]*fyne.MenuItem, 0, len(menu.Items)-1)
 			items = append(items, menu.Items[:i]...)
 			items = append(items, menu.Items[i+1:]...)
-			menu, nextItemID = handleSpecialItems(w, fyne.NewMenu(menu.Label, items...), nextItemID, false)
+			menu.Items = items
```

Comment:
I can see that removing the recursion is a win but now you are modifying the input.
In the end you are modifying the window’s `mainMenu` which is provided by the app developer and I think we should not do that.
So, either we have three options:
- create a new menu for each special item (like the recursion did)
- create a menu copy up front and then modify this (useless copy when there are no special items but this is a setup task, only)
- create a menu copy on demand (probably leads to more complicated code)

Reply:
I reverted to the recursion version after all